src/views/bridge-confirmation/bridge-confirmation.view.tsx

```diff
@@ -188,21 +133,29 @@ const BridgeConfirmation: FC = () => {
               });
             }
           });
-      }
-    };
-    void estimateGas();
-    estimateGasPollInterval.current = window.setInterval(estimateGas, AUTO_REFRESH_RATE);
-  }, [connectedProvider, formData, tokenBalance, callIfMounted, estimateBridgeGas, notifyError]);
+      };
 
-  const stopGasPolling = useCallback(() => {
-    clearInterval(estimateGasPollInterval.current);
-  }, []);
+      connectedProvider.data.provider.on("block", onBlock);
 
-  useEffect(() => {
-    // Start estimating gas
-    startGasPolling();
-    return stopGasPolling;
-  }, [startGasPolling, stopGasPolling]);
+      if (estimatedGas.status === "pending") {
+        onBlock();
+      }
+
+      return () => {
+        connectedProvider.data.provider.off("block", onBlock);
```

Comment:
This unsubscribe is not working. We are subscribing one time per effect execution and never unsubscribing.

Reply:
I've confirmed that the unsubscription works when we do not pass the second param `onBlock`.

Reply:
This smells of an [identity comparison issue](https://github.com/ethers-io/ethers.js/blob/master/packages/providers/src.ts/base-provider.ts#L2186).